frame.js

```diff
@@ -0,0 +1,76 @@
+class Frame {
+
+  constructor() {
+    this.rolls = [];
+    this.scoringRolls = [];
+    this.maxRolls = 2;
+    this.bonus = 0;
+    this.extra = false;
+  }
+
+  roll(pins) {
+    this.rolls.push(pins);
+    this.scoringRolls.push(pins);
+  }
+
+  nonScoringRoll(pins) {
+    this.rolls.push(pins);
+  }
+
+  getRolls() {
+    return [...this.rolls];
+  }
+
+  getScoringRolls() {
+    return [...this.scoringRolls];
+  }
+
+  getTotal() {
+    return this.getRolls().reduce((previousValue, currentValue) => previousValue + currentValue);
+  }
+
+  getScore() {
+    return this.getScoringRolls().reduce((previousValue, currentValue) =>  previousValue + currentValue);
+  }
+
+  isStrike() {
+    return this.getRolls()[0] === 10;
+  }
+
+  isSpare() {
+    return this.getTotal() === 10 && this.numberOfRolls() === this.maxRolls;
+  }
+
+  numberOfRolls() {
+    return this.getRolls().length;
```

Comment:
very interesting I had used an index for this but the length function is a good shout